File: apps/api/src/models/event.model.ts
/* eslint-disable no-use-before-define */
import { BaseModel } from '@itourism/nestjs';

export class EventModel extends BaseModel<EventModel> {
  id: string;

  name: string;

  cityId: string;

  banner: string;

  gallery: string[];

  description: string;

  address: string;

  map: string;

  createdAt: Date;

  updatedAt: Date;
}

File: apps/mobile/lib/modules/events/models/event_model.dart
import 'package:json_annotation/json_annotation.dart';

part 'event_model.g.dart';

@JsonSerializable()
class Event {
  String id;

  String name;

  String cityId;

  String banner;

  List<String> gallery;

  String description;

  String address;

  String map;

  DateTime createdAt;

  DateTime updatedAt;

  Event({
    required this.id,
    required this.name,
    required this.cityId,
    required this.banner,
    required this.gallery,
    required this.description,
    required this.address,
    required this.map,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Event.fromJson(Map<String, dynamic> json) => _$EventFromJson(json);

  Map<String, dynamic> toJson() => _$EventToJson(this);
}
